### ISSUE TITLE
[Feature] Romanization

### ISSUE THREAD

**Issue by faustaahmad** [fast4x/RiMusic#385]
### Description
What's [Romanization](https://wikipedia.org/wiki/Romanization)
Perhaps... if possible, cant read = cant sing
Most of subtitle are on their own language & translator cant do romanization
Even if it's just basic romanization, would be helpful for most overseas ppl.

### Suggested Solution

Ability to convert the text to romaji with couple click
Whether it's by using an extension or included in app
ViMusic aim to be as minimal as possible, but idk much about RiMusic

### Alternatives

Roundabout way...
We know that internet has many free converter to stuff
The problem with free romanization tools that I encountered are added spacing after symbol & random nuked symbol
Also sometimes incorrect romanization, which kinda fine & I can just search for the correct romaji lyrics to compare ig with mine
```
Example: Synchronized Lyrics

[00:00.16]多田葵 - 灼け落ちない翼 (灼热不落的翅膀)
[00:02.04]作词：麻枝准
[00:03.26]作曲：麻枝准
[00:32.17]見渡せる場所に今朝は辿り着く
[00:39.02]遠く線路が伸びる
[00:47.06]こびりついた土かかとから落とし
[00:53.57]歩いた距離を測ってみたり
```
```
Romajidesu

[ 00 : 00 . 16 ] tada mamoru - yake ochinai tsubasa ( 灼热 fu 落的 翅膀 )
[ 00 : 02 . 04 saku 词 ： asa eda jun
[ 00 : 03 . 26 ] sakkyoku ： asa eda jun
[ 00 : 32 . 17 ] miwataseru basho ni kesa wa tadori tsuku
[ 00 : 39 . 02 ] tōku senro ga nobiru
[ 00 : 47 . 06 ] kobiritsuita do kakato kara otoshi
[ 00 : 53 . 57 ] aruita kyori o hakattemi tari
```
See... need to edit line 1 by 1 to manually fix 'em
or can use regex replace, but most of user may not be able to do that in mobile devices
or maybe using a simple script, but it's a hassle to do it

To solve that problem should be very simple, as each timing can only has one line
It would be very helpful to have a button\* that can separate timing with lyrics, so user can easily edit the lyrics themself without worrying about timing
\*Some kind of magic button when editing the lyrics, 1 tap hide timing & user can select all text (without timing & broken line break)

I wish I can do convinient multiline edit on mobile text editor 🤣

### Additional Context

Idk any API nor Library for this kind of task
But when I do seach for it, Wanakana are on the topmost search result but it's written in js & kotlin port last updated 3y ago

**Comment by fast4x** [comment]
Hello, you are italian like me?
"ViMusic aim to be as minimal as possible, but idk much about RiMusic"
If you read wikis or the various tickets you will understand the direction of RiMusic.

Project page has:
RiMusic is a multilingual Android application for streaming music from YouTube Music.
It was **born as an extension of the functionality of ViMusic**, which retains the general appearance, colors.

I’m not sure what you mean, so give me some more detailed examples, what you mean for Romanization.
Thanks

**Comment by faustaahmad** [comment]
I'm asian & not Italian, also hardly ate pizza unless it was my friends or relative treating me 👍

Based from Wikipedia about [Romanization](https://wikipedia.org/wiki/Romanization)
In [linguistics](https://en.m.wikipedia.org/wiki/Linguistics), Romanization or romanisation is the conversion of text from a different [writing system](https://en.m.wikipedia.org/wiki/Writing_system) to the [Roman (Latin) script](https://en.m.wikipedia.org/wiki/Latin_script), or a system for doing so.

I'm afraid if u were motivated to add this feature, it'll cost in app size
I would prefer using alternative method, as is more universal & can solve more problem including bad translation _cough_
"A button to minimize/separate Timing & Lyrics without changing it's Line number"

May get further improved if u can replace those timing text to some kind of individual list & set timer button (like sleep time perhaps) but that may took lots of space in edit lyrics

**Comment by fast4x** [comment]
Very complicated to achieve

**Comment by faustaahmad** [comment]
That's why I recommend alternative solution, but the Title of this issue should be changed into "Improve Edit Lyrics" or something 🤣
<s>so I dont need to manualy edit 'em in dedicated text app that support regex replace to separate 'em manually</s>
```
Edit Lyrics (before)

[00:00.16]多田葵 - 灼け落ちない翼 (灼热不落的翅膀)
[00:02.04]作词：麻枝准
[00:03.26]作曲：麻枝准
[00:32.17]見渡せる場所に今朝は辿り着く
[00:39.02]遠く線路が伸びる
[00:47.06]こびりついた土かかとから落とし
[00:53.57]歩いた距離を測ってみたり
```
```
Edit Lyrics (after)

- Cycle button (All)
[00:00.16]多田葵 - 灼け落ちない翼 (灼热不落的翅膀)
[00:02.04]作词：麻枝准
[00:03.26]作曲：麻枝准
[00:32.17]見渡せる場所に今朝は辿り着く
[00:39.02]遠く線路が伸びる
[00:47.06]こびりついた土かかとから落とし
[00:53.57]歩いた距離を測ってみたり

- Cycle button (Lyrics)
多田葵 - 灼け落ちない翼 (灼热不落的翅膀)
作词：麻枝准
作曲：麻枝准
見渡せる場所に今朝は辿り着く
遠く線路が伸びる
こびりついた土かかとから落とし
歩いた距離を測ってみたり

- Cycle button (Timing)
If possible turn it into list of timer, but this is fine too
OR even if u didnt want to add text only
It's quite easy to modify timing in (All) mode anyway
[00:00.16]
[00:02.04]
[00:03.26]
[00:32.17]
[00:39.02]
[00:47.06]
[00:53.57]
```

**Comment by aneesh1122** [comment]
https://github.com/narVidhai/Google-Transliterate-API

your problem will be solved if this is implemented

**Comment by aneesh1122** [comment]
is it necessary for the code to be in kotlin? I found some working romanization/transliteration repos but they are not in kotlin.

https://github.com/anyascii/anyascii

this is in kotlin but not 100% so I'm not sure.

**Comment by fast4x** [comment]
nice, I'll check it

**Comment by fast4x** [comment]
Implemented, available from version 0.6.60
